### PR TITLE
Fix: User should not be allowed to like/dislike his own posts.

### PR DESCRIPTION
**Pull request by varunlalan**
[Trello ticket](https://trello.com/c/viZQJLRg/32-user-should-not-be-allowed-to-like-dislike-his-own-posts-0)

### CODE REVIEW

**Comment by sampatbadhe** [comment]
:+1: @varunlalan

**Comment by abhijitsinha** [comment]
@varunlalan looks good

File: app/assets/javascripts/like_post.js [hunk 1]
```diff
@@ -1,8 +1,11 @@
 $(document).ready(function(){
 
-  var post_id = $("#post-subject").data('id');
+  var post_id = $("#post-subject").data("id");
+
+  $("body").on("click", ".voterDetails i:not('.voted')", function(event) {
```
Comment (shivanibhanwal) [inline app/assets/javascripts/like_post.js]: @varunlalan do we really need here `body` for the click event. I think we should not check `body` click

File: app/helpers/active_admin/views_helper.rb [hunk 2]
```diff
@@ -4,16 +4,9 @@ def sanitize_data(data)
...
-    else
-      link_to image_tag(
-        "doc.png",
-        title: attachment.attachable_file_name
-        ), attachment.attachable.url, target: "_blank"
-    end
+    image = attachment.is_image? ? attachment.attachable.url(:thumb) : "doc.png"
+
+    link_to image_tag(image, title: attachment.attachable_file_name),
+            attachment.attachable.url, target: "_blank"
```
Comment (shivanibhanwal) [inline app/helpers/active_admin/views_helper.rb]: :+1:

**Comment by shivanibhanwal** [comment]
Other than that `body` click code all looks good.

File: app/assets/javascripts/like_post.js (hunk 1 above)
Comment (varunlalan) [inline app/assets/javascripts/like_post.js]: @shivanibhanwal Yes we dont need to check for `body`.

I see that we used `body` at lots of place. We can remove that too.

**Comment by varunlalan** [comment]
@shivanibhanwal removed `body` from selector check and moved js code to `coffee`. Please review and let me know changes if any. Thanks.

File: app/assets/javascripts/like_post.js.coffee [hunk 3]
```diff
@@ -0,0 +1,21 @@
+$ ->
+
+  $(".voterDetails i").on("click", ->
+    $this = $(@)
+    post_id = $('#post-subject').data('id')
+
+    return if $this.hasClass('voted') or $this.parents('.is_author').length
```
Comment (yogendra689) [inline app/assets/javascripts/like_post.js.coffee]: :+1:

**Comment by rohandaxini** [comment]
merged

**Pull request merged**